src/commands/Minion/chop.ts

```diff
@@ -0,0 +1,101 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { determineScaledLogTime, stringMatches, formatDuration, rand } from '../../lib/util';
+import { BotCommand } from '../../lib/BotCommand';
+
+import { SkillsEnum } from '../../lib/types';
+import { Time, Activity, Tasks } from '../../lib/constants';
+import { WoodcuttingActivityTaskOptions } from '../../lib/types/minions';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import Woodcutting from '../../lib/skills/woodcutting';
+import itemID from '../../lib/util/itemID';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '<quantity:int{1}|name:...string> [name:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, name = '']: [null | number | string, string]) {
+		if (!msg.author.hasMinion) {
+			throw `You dont have a minion`;
+		}
+
+		if (msg.author.minionIsBusy) {
+			return msg.send(msg.author.minionStatus);
+		}
+
+		if (typeof quantity === 'string') {
+			name = quantity;
+			quantity = null;
+		}
+
+		const log = Woodcutting.Logs.find(
+			log => stringMatches(log.name, name) || stringMatches(log.name.split(' ')[0], name)
+		);
+
+		if (!log) {
+			throw `Thats not a valid log to mine. Valid ores are ${Woodcutting.Logs.map(
+				log => log.name
+			).join(', ')}.`;
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Woodcutting) < log.level) {
+			throw `${msg.author.minionName} needs ${log.level} Woodcutting to chop ${log.name}.`;
+		}
+
+		// Calculate the time it takes to mine a single log of this type, at this persons level.
+		let timetoChop = determineScaledLogTime(
+			log!.xp,
+			log.respawnTime,
+			msg.author.skillLevel(SkillsEnum.Woodcutting)
+		);
+
+		// If the user has a dragon axe & over 61 mining provide 10% speed boost
+		if (
```

Comment:
change this how to we check for mining pickaxes, also in line with what we do for mining, give infernal a higher boost.